### PR TITLE
Build: Add extensions to lodash imports

### PR DESCRIPTION
**Pull request by IanVS**
Issue: https://github.com/storybookjs/storybook/issues/20225

## What I did

Added `.js` extension to lodash imports, and turned off `eslint-import/extensions` rule.  There may be other node ESM incompatibilities, but this is a step forwards at least.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: .ci/danger/dangerfile.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ execSync('npm install lodash');
 
 const flatten = require('lodash/flatten');
 const intersection = require('lodash/intersection');
-const isEmpty = require('lodash/isEmpty');
+const isEmpty = require('lodash/isEmpty.js');
```
Comment (valentinpalkovic) [inline .ci/danger/dangerfile.ts:8]: Does the extension make sense here? This is a CommonJS module with CommonJS style imports (require).

Comment (IanVS) [reply .ci/danger/dangerfile.ts:8]: It doesn't hurt anything, and I figured we may as well be consistent.  But I can remove it if preferred.

Comment (valentinpalkovic) [reply .ci/danger/dangerfile.ts:8]: Agreed. Should we then also add the extension to the above two lodash imports?

Comment (IanVS) [reply .ci/danger/dangerfile.ts:8]: I added them there, as well as all the other lodash imports I could find, just for good measure.

**Review by valentinpalkovic — APPROVED** [review]

**Pull request merged**